source/dub/dub.d

```diff
@@ -167,6 +167,47 @@ class Dub {
 
 		init();
 
+		if (skip_registry == SkipPackageSuppliers.none)
+			m_packageSuppliers = getPackageSuppliers(additional_package_suppliers);
+		else
+			m_packageSuppliers = getPackageSuppliers(additional_package_suppliers, skip_registry);
+
+		m_packageManager = new PackageManager(m_dirs.localRepository, m_dirs.systemSettings);
+
+		auto ccps = m_config.customCachePaths;
+		if (ccps.length)
+			m_packageManager.customCachePaths = ccps;
+
+		updatePackageSearchPath();
+	}
+
+	unittest
+	{
+		scope (exit) environment.remove("DUB_REGISTRY");
+		auto dub = new Dub(".", null, SkipPackageSuppliers.configured);
+		assert(dub.m_packageSuppliers.length == 0);
+		environment["DUB_REGISTRY"] = "http://example.com/";
+		dub = new Dub(".", null, SkipPackageSuppliers.configured);
+		logInfo("%s", dub.m_packageSuppliers);
+		assert(dub.m_packageSuppliers.length == 1);
+		environment["DUB_REGISTRY"] = "http://example.com/;http://foo.com/";
+		dub = new Dub(".", null, SkipPackageSuppliers.configured);
+		assert(dub.m_packageSuppliers.length == 2);
+		dub = new Dub(".", [new RegistryPackageSupplier(URL("http://bar.com/"))], SkipPackageSuppliers.configured);
+		assert(dub.m_packageSuppliers.length == 3);
+	}
+
+	/** Get the list of package suppliers.
+
+		Params:
+			additional_package_suppliers = A list of package suppliers to try
+				before the suppliers found in the configurations files and the
+				`defaultPackageSuppliers`.
+			skip_registry = Can be used to skip using the configured package
+				suppliers, as well as the default suppliers.
+	*/
+	private PackageSupplier[] getPackageSuppliers(PackageSupplier[] additional_package_suppliers, SkipPackageSuppliers skip_registry)
```

Comment:
if this is well documented and unittested I think it would make sense to make this public

Reply:
done :)